tensorflow/compiler/jit/xla_compilation_cache.cc

```diff
@@ -410,14 +548,17 @@ Status XlaCompilationCache::CompileImpl(
   // cache eviction.
   mutex_lock entry_lock(entry->mu);
   int64 current_request_count = ++entry->request_count;
-  VLOG(2) << "Compilation cache entry hit: " << entry->compiled
-          << " signature: " << signature.HumanString() << " with request count "
+  VLOG(2) << "Compilation cache entry hit: "
+          << static_cast<int>(entry->compile_state)
+          << " signature: " << human_signature << " with request count "
           << current_request_count << " and compile threshold "
           << compile_threshold.value_or(0);
-  if (!entry->compiled) {
+  bool return_null = false;
+  CompileState state = entry->compile_state;
+  if (state == CompileState::kUncompiled) {
```

Comment:
Can you please add a `// TODO(sanjoy): Refactor this code into helper functions.` (or send a PR :) ).  The flow has here has become quite difficult to follow.

Reply:
Not addressed?

Reply:
done